lib/decanter/parser/date_parser.rb

```diff
@@ -1,16 +1,21 @@
 module Decanter
   module Parser
     class DateParser < ValueParser
-
       allow Date
 
       parser do |val, options|
-        raise Decanter::ParseError.new 'Expects a single value' if val.is_a? Array
-        next if (val.nil? || val === '')
+        raise Decanter::ParseError, 'Expects a single value' if val.is_a? Array
+        next if val.nil? || val === ''
+
         parse_format = options.fetch(:parse_format, '%m/%d/%Y')
-        ::Date.strptime(val, parse_format)
+        begin
+          ::Date.strptime(val, parse_format)
+        rescue ArgumentError => e
+          raise Decanter::ValueFormatError, 'invalid Date value for format' if e.message == 'invalid date'
+
+          raise ArgumentError, e.message
```

Comment:
Can you walk me through the rationale for raising a new error, versus the existing one? i.e., (`raise e`)

It seems like it potentially introduces some misdirection with a longer call stack
![Image 2023-03-31 at 11 00 41 AM](https://user-images.githubusercontent.com/20426095/229157005-33e643cb-7117-4635-9461-e8922029a5ac.jpg)